### PR TITLE
require dart 3.0; rev to 0.4.1 in prep for publishing

### PR DESCRIPTION
**Pull request by devoncarew**
- require dart 3.0
- rev to 0.4.1 in prep for publishing
- update to the latest version of `dart_flutter_team_lints`
- fix a link to the CI status in the readme

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,12 @@
+## 0.4.1
+
+- Require Dart 3.0.
+- Update to `package:dart_flutter_team_lints` 2.0.
```
Comment (kevmoo) [inline CHANGELOG.md:4]: not sure visible. Remove?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,12 @@
+## 0.4.1
+
+- Require Dart 3.0.
```
Comment (kevmoo) [inline CHANGELOG.md:3]: Might could just bump it to 3.2 – and get new inference magic!

**Review by kevmoo — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (devoncarew) [reply CHANGELOG.md:3]: sgtm - will do

File: CHANGELOG.md (hunk 1 above)
Comment (devoncarew) [reply CHANGELOG.md]: sgtm - will do

**Pull request merged**